node/src/actors/chain_manager/handlers.rs

```diff
@@ -135,13 +129,14 @@ impl Handler<EpochNotification<EveryEpochPayload>> for ChainManager {
                         // Persist block and update ChainState
                         self.consolidate_block(ctx, &block, utxo_diff);
                     } else {
+                        let previous_epoch = msg.checkpoint - 1;
                         log::warn!(
                             "There was no valid block candidate to consolidate for epoch {}",
-                            msg.checkpoint - 1
+                            previous_epoch
                         );
 
                         // Update ActiveReputationSet in case of epochs without blocks
-                        if let Err(e) = rep_engine.ars.update(vec![], expected_ars_epoch) {
+                        if let Err(e) = rep_engine.ars.update(vec![], previous_epoch) {
```

Comment:
(2/2) ...here

Reply:
the problem are empty epochs while we are consolidating old blocks. For example:
We have a very few probability to mine, and there are only one miner now. After some empty epochs, the reputation engine is updated and now that miner has 100% of the probability and mine a block. If later other node has to synchronize that chain, if we dont update the reputation engine before to check the poe, this node keeps the few probability until it consolidate a new block, so the PoE that the other node show (f555... for example) is not valid for it, and reject a valid block thinking that it is not

Reply:
So it is a problem while synchronizing? Then we can handle it in `AddBlocks`

Reply:
:thinking: good point, I am going to try